staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -358,6 +364,23 @@ func (lsel internalSelector) String() string {
 	return strings.Join(reqs, ",")
 }
 
+// RequiresExactMatch introspect whether a given selector requires a single specific field
+// to be set, and if so returns the value it requires.
+func (lsel internalSelector) RequiresExactMatch(label string) (value string, found bool) {
+	for ix := range lsel {
+		if lsel[ix].key == label {
+			switch lsel[ix].operator {
+			case selection.Equals, selection.DoubleEquals, selection.In:
```

Comment:
If all operators are selection.In and strValues have multiple values, it seems the func can short circuit and return false directly.